grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -349,11 +351,35 @@ class QueryService implements Status {
                 .setString('consentKey', consentKey)
                 .list()
         results.collect {
-            sampleInfo.put(it, StorageDocument.findAllByConsentCollectionLinkId(it.id))
+            consentCollectionIds.add(it.id)
+            consentCollectionLinkList.add(it)
+        }
+        List <StorageDocument> storageDocuments = findAllDocumentsBySampleCollectionId(consentCollectionIds)
+        consentCollectionLinkList.collect {
+            sampleInfo.put(it,
+                    storageDocuments.findAll{
+                    doc -> doc.consentCollectionLinkId == it.id
+                }
+            )
         }
         sampleInfo
     }
 
+    List <StorageDocument> findAllDocumentsBySampleCollectionId(List <Integer> consentCollectionIds) {
```

Comment:
What I think you really want here is not a list of storage documents. You want a map of consent collection link id -> list of storage documents. That will allow you to get rid of the nested query above. You can easily do this with the collection `groupBy` convenience method.

Reply:
Great idea. I had to cast a value to BigInteger here to match types from the place this function is called.